compiler/rustc_ty_utils/src/ty.rs

```diff
@@ -193,7 +193,7 @@ fn adt_sized_constraint(tcx: TyCtxt<'_>, def_id: DefId) -> ty::AdtSizedConstrain
     let result = tcx.mk_type_list(
         def.variants
             .iter()
-            .flat_map(|v| v.fields.last())
+            .flat_map(|v| &v.fields)
```

Comment:
So, this isn't needed (i.e. this isn't a bug). We know that all fields except the last for all variants must be `Sized` for the struct to be well formed. (This check is done elsewhere.)